generated-code-samples-for-review/speech/speech_transcribe_async.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by protoc-gen-go_gapic. DO NOT EDIT.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	speech "cloud.google.com/go/speech/apiv1"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1"
+)
+
+// [START speech_transcribe_async]
+
+// sampleLongRunningRecognize: Transcribe a long audio file using asynchronous speech recognition
+//
+// localFilePath: Path to local audio file, e.g. /path/audio.wav
+func sampleLongRunningRecognize(localFilePath string) error {
+	ctx := context.Background()
+	c, err := speech.NewClient(ctx)
+	if err != nil {
+		return err
```

Comment:
We usually include additional context in errors to say what it was that caused the error. Usually, the function name that returned the error is enough (judgement call to include the package name or other stuff before the function name).

```go
return fmt.Errorf("speech.NewClient: %v", err)
```

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#return-errors

Reply:
Makes sense! Just need to change the template!